src/devices/Ina219/Ina219.cs

```diff
@@ -0,0 +1,375 @@
+
+            regValue &= (ushort)ConfigurationFlags.BadcMask ^ 0xFFFF;
+            regValue |= (ushort)((ushort)resorsamp << 4);
+
+            WriteRegister(Register.Configuration, regValue);
+
+            _busAdcResSamp = resorsamp;
+        }
+
+        /// <summary>
+        /// Set the Ina219 ADC resolution or samples to be used when reading the Shunt voltage.
+        /// </summary>
+        /// <remarks>
+        /// This can either by the number of bits used for the ADC conversion (9-12 bits) or the number of samples at 12 bits to be averaged for the result.
+        /// </remarks>
+        /// <param name="resorsamp">The ADC resolution or sample count to switch to.</param>
+        public void SetShuntAdcResolutionOrSamples(AdcResolutionOrSamples resorsamp)
+        {
+            ushort regValue = ReadRegister(Register.Configuration);
+
+            regValue &= (ushort)ConfigurationFlags.SadcMask ^ 0xFFFF;
+            regValue |= (ushort)resorsamp;
+
+            WriteRegister(Register.Configuration, regValue);
+
+            _shuntAdcResSamp = resorsamp;
+        }
+
+        /// <summary>
+        /// Set the Ina219 calibration value used to scale the Shunt voltage into a Current reading.
+        /// </summary>
+        /// <remarks>
+        /// If this method is used then the values for the CurrentLsb and PowerMsb will have to be manually determined
+        /// to ensure accurate readings from GetPower and GetCurrent Methods.
+        /// <see cref="GetPower"/><seealso cref="GetCurrent"/><seealso cref="PowerLsb"/><seealso cref="CurrentLsb"/>
+        /// </remarks>
+        /// <param name="calibrationValue">The calibration value to switch to.</param>
+        public void SetCalibration(ushort calibrationValue)
+        {
+            _calibrationValue = calibrationValue;
+            WriteRegister(Register.Calibration, calibrationValue);
+        }
+
+        /// <summary>
+        /// Set the Ina219 calibration values used to scale the Shunt voltage into a Current reading based on the sensitivity of the
+        /// Progmamable Gain Amplifier, the required full scale current and the value of the shunt resistor.
+        /// </summary>
+        /// <remarks>
+        /// This method sets the calibration register and also the PowerLsb and CurrentLsb properties.
+        /// <see cref="PowerLsb"/><seealso cref="CurrentLsb"/>
+        /// </remarks>
+        /// <param name="sensitivity">The sensitivity of the shunt Programable Gain Amplifier</param>
+        /// <param name="fsCurrent">The required full scale current value</param>
+        /// <param name="shuntResistance">The value of the shunt resistor. This defaults to 0.1 Ohms as used in the Adafruit breakout board.</param>
+        public void SetCalibration(PgaSensitivity sensitivity, float fsCurrent, float shuntResistance = 0.1F)
+        {
+            int i;
+            float minCurrentLsb = fsCurrent / 32767;
```

Comment:
where does this math in this function coming from? (please datasheet page and link if it's not there) I can see this is slightly over complicated but I wanted double check this for correctness.

I.e. I know:

```csharp
while (i * currentStep < minCurrentLsb) { i++; };
CurrentLsb = i * currentStep;
```

will be equivalent to something along the lines of (not tested):

```csharp
CurrentLsb = currentStep * Math.Floor(minCurrentLsb / currentStep);
```

Reply:
I have removed the auto calibration method and detailed an example of calibration in the sample